prepare_data/src/main/groovy/org/orbisgis/osm/FormattingForAbstractModel.groovy

```diff
@@ -137,3 +177,65 @@ static int getNbLevels (def row) {
     return result
 }
 
+// define the value of the column width according to the value of width from OSM
+static Float getWidth (String width){
+    Float result
+    if (width != null && width.isFloat()) {
+        result=width.toFloat()
+    }
+    return result
+}
+
+// define the value of the column width according to the value of width from OSM
+static Integer getZIndex (String zindex){
+    Integer result
+    if (zindex != null && zindex.isInteger()) {
+        result=zindex.toInteger()
+    }
+    return result
+}
+// Get the equivalent abstract value for a set of osm key/value elements
+static String getAbstractValue(def row, def myMaps) {
+    String strType = null
+    myMaps.each { finalVal ->
+        finalVal.value.each { osmVals ->
+            osmVals.value.each { osmVal ->
+                def val = osmVal.toString()
+                if (val.startsWith("!")) {
+                    val.replace("! ","")
+                    if ((row.getString(osmVals.key) != val) && (row.getString(osmVals.key) != null)) {
+                        if (strType == null) {
+                            strType = finalVal.key
+                        }
+                    }
+                } else {
+                    if (row.getString(osmVals.key) == val) {
+                        if (strType == null) {
+                            strType = finalVal.key
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return strType
+}
+
+// define the value of the column sidewalk according to the values of sidewalk from OSM
```

Comment:
To document the method, use javadoc comment `/**` instead of `//`

Reply:
Done